PolyEngine/Engine/Res/Shaders/lightAccumulation.frag.glsl

```diff
@@ -130,38 +128,35 @@ vec3 fresnelSchlickRoughness(float cosTheta, vec3 F0, float roughness)
     return F0 + (max(vec3(1.0 - roughness), F0) - F0) * pow(clamp(1.0 - cosTheta, 0.001, 1.0), 5.0); // clamp to prevent NaNs
 }
 
-float calcShadow(vec4 fragPosInDirLight, float NdotL)
+#pragma include "pcf.inc.glsl"
+#pragma include "evsm.inc.glsl"
+
+uniform int uShadowType;
+
+float calculateShadow(vec4 fragPosInDirLight)
 {
-	// perform perspective divide
-    // vec3 projCoords = fragPosInDirLight.xyz / fragPosInDirLight.w;
-    vec3 projCoords = fragPosInDirLight.xyz;
-	// transform to [0,1] range
-    projCoords = projCoords * 0.5 + 0.5;
-
-    if (projCoords.z > 1.0)
-        return 1.0;
-
-    // get closest depth value from light's perspective (using [0,1] range fragPosLight as coords)
-    float closestDepth = texture(uDirShadowMap, projCoords.xy).r;
-    // get depth of current fragment from light's perspective
-    float currentDepth = projCoords.z;    
-    // check whether current frag pos is in shadow
-	float bias = max(0.05 * (1.0 - NdotL), 0.005);
-	
-    float shadow = 0.0;
-    vec2 texelSize = 1.0 / textureSize(uDirShadowMap, 0);
-    for (int x = -1; x <= 1; ++x)
-    {
-        for (int y = -1; y <= 1; ++y)
-        {
-            float pcfDepth = texture(uDirShadowMap, projCoords.xy + vec2(x, y) * texelSize).r;
-            shadow += currentDepth - bias > pcfDepth ? 0.0 : 1.0;
-        }
-    }
-    shadow /= 9.0;
-    
+    vec3 posInLight = fragPosInDirLight.xyz / fragPosInDirLight.w; // ClipSpace to NDC
+    posInLight = posInLight * 0.5 + 0.5; // NDC to [0, 1]
 
-    return shadow;
+	// TODO: use defines
```

Comment:
use at least some constants at the beggining of file. Magic numbers are a bad idea.

Reply:
Created contants